src/contract/assignments.rs

```diff
@@ -150,6 +151,25 @@ impl ToMerkleSource for OwnedRightsInner {
     }
 }
 
+impl MergeRevealed for OwnedRights {
```

Comment:
Why we have it here while the rest of implementations are in a standalone file `merge.rs`?

Reply:
This is because `OwnedRightsInner` has `pub(super)` visibility defined. Because I am using `OwnedRightsInner` to check the consistency of two `OwnedRights`, it had to be implemented in the `assignment.rs`.

If we can change the visibility of `OwnedRightsInner` to `pub`, then we can move this to `merge.rs`.